cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,702 @@
+
+foreach(name ${pm_var_names})
+  set_property(
+    TARGET partition_manager
+    PROPERTY ${name}
+    ${${name}}
+    )
+endforeach()
+
+# Turn the space-separated list into a Cmake list.
+string(REPLACE " " ";" PM_ALL_BY_SIZE ${PM_ALL_BY_SIZE})
+
+# Iterate over every partition, from smallest to largest.
+foreach(part ${PM_ALL_BY_SIZE})
+  string(TOUPPER ${part} PART)
+  get_property(${part}_PM_HEX_FILE GLOBAL PROPERTY ${part}_PM_HEX_FILE)
+  get_property(${part}_PM_ELF_FILE GLOBAL PROPERTY ${part}_PM_ELF_FILE)
+
+  # Process container partitions (if it has a SPAN list it is a container partition).
+  if(DEFINED PM_${PART}_SPAN)
+    string(REPLACE " " ";" PM_${PART}_SPAN ${PM_${PART}_SPAN})
+    list(APPEND containers ${part})
+  endif()
+
+  # Include the partition in the merge operation if it has a hex file.
+  if(DEFINED ${part}_PM_HEX_FILE)
+    get_property(${part}_PM_TARGET GLOBAL PROPERTY ${part}_PM_TARGET)
+    list(APPEND explicitly_assigned ${part})
+  else()
+    if(${part} IN_LIST images)
+      get_shared(${part}_bin_dir  IMAGE ${part} PROPERTY ZEPHYR_BINARY_DIR)
+      get_shared(${part}_hex_file IMAGE ${part} PROPERTY KERNEL_HEX_NAME)
+      get_shared(${part}_elf_file IMAGE ${part} PROPERTY KERNEL_ELF_NAME)
+      set(${part}_PM_HEX_FILE ${${part}_bin_dir}/${${part}_hex_file})
+      set(${part}_PM_ELF_FILE ${${part}_bin_dir}/${${part}_elf_file})
+      set(${part}_PM_TARGET ${part}_subimage)
+    elseif(${part} IN_LIST containers)
+      set(${part}_PM_HEX_FILE ${PROJECT_BINARY_DIR}/${part}.hex)
+      set(${part}_PM_TARGET ${part}_hex)
+    endif()
+    list(APPEND implicitly_assigned ${part})
+  endif()
+endforeach()
+
+if (${is_dynamic_partition_in_domain})
+  set(merged_suffix _${DOMAIN})
+  string(TOUPPER ${merged_suffix} MERGED_SUFFIX)
+endif()
+set(merged merged${merged_suffix})
+set(MERGED MERGED${MERGED_SUFFIX})
+
+set(PM_${MERGED}_SPAN ${implicitly_assigned} ${explicitly_assigned})
+set(${merged}_overlap TRUE) # Enable overlapping for the merged hex file.
+
+# Iterate over all container partitions, plus the "fake" merged paritition.
+# The loop will create a hex file for each iteration.
+foreach(container ${containers} ${merged})
+  string(TOUPPER ${container} CONTAINER)
+
+  # Prepare the list of hex files and list of dependencies for the merge command.
+  foreach(part ${PM_${CONTAINER}_SPAN})
+    string(TOUPPER ${part} PART)
+    list(APPEND ${container}hex_files ${${part}_PM_HEX_FILE})
+    list(APPEND ${container}elf_files ${${part}_PM_ELF_FILE})
+    list(APPEND ${container}targets ${${part}_PM_TARGET})
+  endforeach()
+
+  # Do not merge hex files for empty partitions
+  if(NOT ${container}hex_files)
+    list(REMOVE_ITEM PM_${MERGED}_SPAN ${container})
+    continue()
+  endif()
+
+  # If overlapping is enabled, add the appropriate argument.
+  if(${${container}_overlap})
+    set(${container}overlap_arg --overlap=replace)
+  endif()
+
+# Should all files still be merged ?
```

Comment:
It's useful to have a final hex file that you can use for production, but also it completely deviates from the zephyr style. It was also annoying to use (not sure if it's changed since) but you had to build an application, then flash it **with a connected device** in order to get a set of hex files, if there was no connected device then it wouldn't generate them.

Reply:
> It's useful to have a final hex file that you can use for production, 

I agree, it may be useful in many cases, but in such a case we should make it a clear feature / a feature with a purpose.
One such feature could be for production systems, in which case the following must be discussed in terms of how the feature should look:
- Ability to build source and create hex-file without the need to flash
- Should everything be merged, or kept in flashable hex files / domains   
   - Example, the merged hex for nRF5340 cannot be flashed directly but must be split into two files, one for each core.
      In such case, perhaps the best is that a merged hex is created per core.

The merged hex feature should not be just a side-effect of how other things were previously done.